### PR TITLE
Fetch catalog.xml on server and provide datasets for client by /datasets route

### PR DESCRIPTION
**Pull request by maxbischof**
Curated datasets in settings.js is removed by th fetch of catalog.xml in server.js. Metadata about available datasets is provided by /datasets route than. With this update all csv datasets from the open data sh platform are available on the platform. 

Deployed [here](https://data-hub-nor-catalog-kqrxp3dji.herokuapp.com/) (Could take a few minutes to load, due to sleeping dyno)

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,24 +1,41 @@
...
-          path={`/datensaetze/${dataset.name.replace(' ', '-')}-${dataset.id}`}
-          render={() => <DatasetDetailsPage dataset={dataset} />}
-        />
-      ))}
+      {datasets ? (
+        <>
+          <Route
+            exact
+            path="/"
+            render={() => <RootPage datasets={datasets} />}
```
Comment (karstenlenz) [inline client/src/App.js:26]: Why do you  use `render={...}` as a prop? I think you could just wrap the rootpage between the route tags, if you like like this: 

`<Route> <Rootpage /> </Route>` (need to add the props of course)

File: client/src/components/DatasetDetails.js [hunk 2]
```diff
@@ -47,8 +52,7 @@ export default function DatasetDetails({
 }
 
 DatasetDetails.propTypes = {
```
Comment (karstenlenz) [inline client/src/components/DatasetDetails.js:54]: I think it is more helpful for other developers to have the proptypes at the top so you can easily understand what kind of props the component requires.

**Review by karstenlenz — APPROVED** [review]
Looks great. I didn't find much to improve and just left 2 minor comments.

File: client/src/components/DatasetDetails.js (hunk 2 above)
Comment (maxbischof) [reply client/src/components/DatasetDetails.js:54]: Saw that you doing it on top. But I dont like it there. The actual logic of the component is more important to me.

File: client/src/App.js (hunk 1 above)
Comment (maxbischof) [reply client/src/App.js]: Oh yes good question. Thanks. Router documentation is pretty clear on that topic: 

"The recommended method of rendering something with a <Route> is to use children elements, as shown above. There are, however, a few other methods you can use to render something with a <Route>. These are provided mostly for supporting apps that were built with earlier versions of the router before hooks were introduced."

Now passing the components as children.

**Pull request merged**